### DISCUSSION TITLE
Cannot add hostname -> system crashes

### DISCUSSION THREAD

**Discussion by phanthomias** [globaleaks/globaleaks-whistleblowing-software#3974] [General]
Hello,
I've installed Globaleaks 4.14.7 on a Ubuntu 22.04.03 LTS server and it works fine when I access it with the IP address.
Now I would like to add the hostname. Under Network -> HTTPS there is no entry at all (not even for localhost as I have seen it on a screenshot in the documentation:
![grafik](https://github.com/globaleaks/GlobaLeaks/assets/157718779/dd799df5-50b6-4854-ace1-3d7b2a9499c3)

When I enter a hostname and click save, I cannot see any result. The only thing that happens is, that I get an error message in the upper left corner and I totally screw the system. I cannot login anymore and when filing a report the same error appears:
![GL-6](https://github.com/globaleaks/GlobaLeaks/assets/157718779/0fe74755-823f-4fba-808e-20a1e0c3dd25)
![2](https://github.com/globaleaks/GlobaLeaks/assets/157718779/49e11206-eb07-4c43-8ac5-1333442aead6)

In the log file, I see that the url has been added but it is actually not working to access the system with this URL and the error always appears which makes it impossible to login and use the system:
![GL-4](https://github.com/globaleaks/GlobaLeaks/assets/157718779/e28daed1-cfe3-49c0-8fea-3b1f60e11c06)

I have to restore the backup of the server from before the change in order to use it again.

Can anyone help with this? Where are the configurations that are made in the backend saved? Is it possible to edit a local config file on the server also?

Thanks and best regards

**Comment by evilaliv3** [comment]
Hello.

Configurations are saved on the database in the table 'config', in the variable 'hostname' on the file /var/globaleaks/glboaleaks.db

If you want you can reach me privately on community.globaleaks.org and share details about the specific hostname format so that we could debug together the issue.

**Reply by phanthomias** [reply]
Hello,
could you please help me with my issue, so that we can actually use GlobaLeaks?
Thanks and best regards

**Comment by phanthomias** [comment]
Hi,
thanks for your quick reply 👍 
I think I've tried to add the host in the format "subdomain.mydomain.com" and in the logfile it appears "https://subdomain.mydomain.com".
It might also be that I have entered "https://subdomain.mydomain.com" directly. Which format should be entered here? 
Since the system totally crashes afterwards and I don't know how to fix it besides restoring the backup of the VM, I 'd rather not try it again.
Is it normal, that I cannot see any data here, not even localhost or the local IP of the server? 
Can you help me with this issue?

Thanks and best regards,
Minh

**Comment by evilaliv3** [comment]
I've actually doubts that the issue is related to the software.

Please feel free to reach me on community.globaleaks.org and pass me the hostname so to verify together.

**Comment by MIAGGmbH** [comment]
Hi,
i tried to install and configure GlobaLeaks today. Unfortunately, I'm facing the same issue. As soon as I enter the hostname report.company.tld, the system crashes. Is there already a solution? The installation was performed on Ubuntu 22.04.

**Comment by evilaliv3** [comment]
Thank you @MIAGGmbH 

May i ask you what you mean with "the system crashes"?

**Reply by MIAGGmbH** [reply]
Once I enter the hostname report.miag.de and press save, I only receive HTTP 400 messages. There is nothing in the log. Apparently, GlobaLeaks starts, but it is not accessible.

**Comment by daniele338** [comment]
Hi, 
I encountered the same problem of @MIAGGmbH. 
Only fix available was to reinstall the software. 

The domain name used is format subdomain.domain.root_domain

No error returned inside the logs.

OS Used: Debian 12

**Comment by MIAGGmbH** [comment]
The issue persists after updating to version 4.15.
![grafik](https://github.com/globaleaks/GlobaLeaks/assets/162584044/ca4e7b2b-6be4-4a7f-ae79-8e100ac4a3b2)

**Reply by evilaliv3** [reply]
Is this a single tenant setup or a multi tenant setup?

Do you get the error clicking save after configuring the hostname or while activating the HTTS certificate?

**Reply by MIAGGmbH** [reply]
its a single tenant setup
I get the error when i set the hostname and click save. With klick on save, the system isnt reachable anymore. 
Only reinstall or going back with snapshot fix it. 
I've tried Ubuntu 20.04 & 22.04, Debian 10, 11, 12.
On all systems the same error...

**Reply by evilaliv3** [reply]
Thank you. I have actually just setup a fresh system and tried to configure the hostname 'report.miag.de' and i'm actually not able to reproduce the bug.

Is this the exact domain that you are configuring? that would be very helpful, thank you.

Are you able to record a small video of your process featuring the exact hostname and clicking on the save button ?

**Reply by MIAGGmbH** [reply]
Yes, this is the exact domain. 
I install now a fresh Ubuntu 22.04 again and record a video from GlobaLeaks install and configure.

**Reply by evilaliv3** [reply]
Thank you.

This is actually very strange.

Thank you if you could as well upload:
/var/globaleaks/log/access.log
/var/globaleaks/log/globaleaks.log
the output of the command dmesg after issuing the save command

**Reply by MIAGGmbH** [reply]
Wow... i've recorded the installation and configuration... but now it works... 
the one and only change is now ubuntu 22.04 in english. Not in German... 
I try to reproduce the bug on german systems today at home and give a feedback. 
@daniele338 maybe its relevant for you..

Thank u for your time @evilaliv3

**Reply by daniele338** [reply]
@MIAGGmbH I tried right now by insterting a random string (not a domain) inside the hostname field.
It's odd.
But I tried it on a debian 12.

**Reply by daniele338** [reply]
@MIAGGmbH sorry for the doble message. 
But do you have the possibilities to check the version of python used?
Have you used the commands adviced to install the product?
```

wget https://deb.globaleaks.org/install-globaleaks.sh
chmod +x install-globaleaks.sh
./install-globaleaks.sh
```

**Reply by MIAGGmbH** [reply]
@daniele338 hi. yeah i used the commands to install. 
Have u tried the english systemlanguage? Now it works for me with my own domain.
